### PR TITLE
chore: update carbon and add scss prefix

### PR DESCRIPTION
**Pull request by lee-chase**
Updated Carbon and configs to remove prefix --ibm requirement on CSS variables.

Adds 'exp' prefix to SCSS in example component.

### CODE REVIEW

**Review by davidmenendez — COMMENTED** [review]
looks good, but I just have a few comments

**Comment by SimonFinney** [comment]
@lee-chase I'm curious to know what the 'exp' prefix is signifying here - Is this a standard we plan to use moving forward or is it purely for example purposes?

**Comment by lee-chase** [comment]
> @lee-chase I'm curious to know what the 'exp' prefix is signifying here - Is this a standard we plan to use moving forward or is it purely for example purposes?

Purely example, the CDAI components all have a prefix. I think a prefix can be useful, but should be based on core package name.

**Comment by matthew-chirgwin** [comment]
> @lee-chase I'm curious to know what the 'exp' prefix is signifying here - Is this a standard we plan to use moving forward or is it purely for example purposes?

To @lee-chase's point, we have had cases in CDAI where product A has version X of the components/styles, and product B has version Y, and those styles clash, causing defects/poor UX. The prefix can be set/used to avoid such clashes. Thus having one (with a sensibly named default value) is definitely helpful, and easier to set early than retrofit later. I agree with Lee that said prefix should be defined and used as a constant from the core package.

What the prefix is name/value wise is a whole other question however :)

File: packages/experimental-package/src/global/js/settings.js [hunk 1]
```diff
@@ -0,0 +1,3 @@
+const expPrefix = 'exp';
```
[the remark this answers is not in the document]
Comment (lee-chase) [reply packages/experimental-package/src/global/js/settings.js:1]: Was rather copying carbon and https://github.ibm.com/cloud-integration-design/ide-components

**Comment by lee-chase** [comment]
@matthew-chirgwin seen any better way to set a css var? This seems a bit excessive, but I couldn't get a direct object assignment to work with a prefix.

```
  const [exampleStyle, setExampleStyle] = useState({});

  useEffect(() => {
    setExampleStyle({
      [`--${expPrefix}-border-color`]: borderColor,
    });
  }, [borderColor]);

  // ...
  style={exampleStyle}
```

**Comment by matthew-chirgwin** [comment]
> @matthew-chirgwin seen any better way to set a css var? This seems a bit excessive, but I couldn't get a direct object assignment to work with a prefix.
> 
> ```
>   const [exampleStyle, setExampleStyle] = useState({});
> 
>   useEffect(() => {
>     setExampleStyle({
>       [`--${expPrefix}-border-color`]: borderColor,
>     });
>   }, [borderColor]);
> 
>   // ...
>   style={exampleStyle}
> ```

I assume you tried something like this with no success?:

```
...
  <div style={{[`--${expPrefix}-border-color`]: borderColor}}>
      ...
  </div>
...
```

`--` is not a valid object key, but `'--'` is. I wonder if that is a factor?

**Comment by lee-chase** [comment]
> {{[`--${expPrefix}-border-color`]: borderColor}}

I did but I was being a bit more useless and that with

```style={border && { [`--${expPrefix}-border-color`]: borderColor}}```

When what I really wanted was

```style={{ [`--${expPrefix}-border-color`]: borderColor ? borderColor : undefined}}```

However, it turns out that I didn't need to test as there is no default for that prop

```style={{ [`--${expPrefix}-border-color`]: borderColor}}```

Sanity restored.

File: packages/experimental-package/src/global/js/settings.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (davidmenendez) [reply packages/experimental-package/src/global/js/settings.js:1]: interesting i was unaware of that. if that's the standard carbon already set i'm fine with keeping it.

**Review by davidmenendez — APPROVED** [review]
thanks for reviewing my comments. looks like based on various feedback this PR is in good standing. approving 👍

**Pull request merged**